### ISSUE TITLE
session_start() issue

### ISSUE THREAD

**Issue by dev-101** [piwik/piwik#11491]
Hi, I am using an auto-prepend script which requires sessions for proper operation.
This conflicts with Piwik, which also tries to start own session, generating notices with each request, filling up the logs quickly (and most importantly, can't log in or access reports).

My question is how hard would it be to implement a simple logic that checks if the session already exists, e.g.:

`if(!isset($_SESSION)) session_start();`

I have read many older topics, and I am aware of the alternative to store sessions into db, which I do not really want. I wonder why is this so hard to implement - do you have a particular reason for it?

Thanks

**Comment by sgiehl** [comment]
Why don't you try to disable you auto-prepend file using .htaccess? Or aren't you using Apache?

**Comment by dev-101** [comment]
Hi, well, yeah, there is that option as well, I have root access.
But, I was just wondering - why is this the way it is?
And auto session start can sometimes come in handy, so I would like to keep it if possible.

**Comment by mattab** [comment]
Thanks for the report. Feel free to open a pull request, otherwise we won't fix this ourselves.

**Comment by dev-101** [comment]
Hi mattab,

I looked into the source code of Piwik, and for one, I could not locate where the hack it starts the session?

It does not use session_start() method, for sure. I have located Session class and traced the code from there, without any luck.

Also, I see that Piwik 'likes' to store it's session files to local folders, instead of general one (defined in php.ini).

Now, I do not understand why is your attitude towards this issue so strict, because, sessions are shared resources in PHP, it is not something that can be taken for granted. If any app could start it's own session (by unique ID), we would not have such a problem. But, that is not the case, so, a reasonable and responsible use should be conducted IMHO.

Thanks,
Regards

**Comment by dev-101** [comment]
P.S.
I have tried the alternative method with database storage, but for some reason, it did not work.
I have edited config file and added the line per instructions under [General] section, both:

`session_save_handler = dbtable`

and

`session_save_handler = "dbtable"`

Do we have to create new table @ piwik's database with particular name and replace generic "dbtable" from above? Because, above line just does not work at all.
